eZ/Bundle/EzPublishCoreBundle/Routing/JsRouting/ExposedRoutesExtractor.php

```diff
@@ -43,7 +45,7 @@ public function getRoutes()
     public function getBaseUrl()
     {
         $baseUrl = $this->innerExtractor->getBaseUrl();
-        $siteAccess = $this->masterRequest->attributes->get('siteaccess');
+        $siteAccess = $this->requestStack->getMasterRequest()->attributes->get('siteaccess');
```

Comment:
We should check if `Request` is even present in `RequestStack`.

```suggestion
        $request = $this->requestStack->getMasterRequest();
        if ($request === null) {
            return $baseUrl;
        }
        
        $siteAccess = $request->attributes->get('siteaccess');
```

On a side note, it probably should have used `RequestStack::getCurrentRequest()` instead of `RequestStack::getMasterRequest()`, but I'm unsure about it, so... we should stick with the current variant I guess?

Reply:
Also note that the `master` contains the following code in the `getBaseUrl()` method.

```php
    public function getBaseUrl(): string
    {
        $masterRequest = $this->requestStack->getMasterRequest();
        $baseUrl = $this->innerExtractor->getBaseUrl();
        $siteAccess = $masterRequest->attributes->get('siteaccess');
        if ($siteAccess instanceof SiteAccess && $siteAccess->matcher instanceof SiteAccess\URILexer) {
            $baseUrl .= $siteAccess->matcher->analyseLink('');
        }

        return $baseUrl;
    }
```

I assume we want to merge it up immediately, so it should either be roughly the same, or immediately replaced?

Reply:
> We should check if `Request` is even present in `RequestStack`.
> 
> On a side note, it probably should have used `RequestStack::getCurrentRequest()` instead of `RequestStack::getMasterRequest()`, but I'm unsure about it, so... we should stick with the current variant I guess?

It's not that obvious for me. What if non-master request is skipped intentionally? Not sure what are functional requirements for JsRouting. Feedback on that welcomed.

Reply:
> > We should check if `Request` is even present in `RequestStack`.
> > On a side note, it probably should have used `RequestStack::getCurrentRequest()` instead of `RequestStack::getMasterRequest()`, but I'm unsure about it, so... we should stick with the current variant I guess?
> 
> It's not that obvious for me. What if non-master request is skipped intentionally? Not sure what are functional requirements for JsRouting. Feedback on that welcomed.

On second thought... Maybe variable name was misleading and what was injected as `Request` was always current request?

Reply:
> On second thought... Maybe variable name was misleading and what was injected as Request was always current request?

Based on service declaration (`@=service("request_stack").getMasterRequest()`) master request was injected.

I'm only considering those for ESI includes and similar, as that's the only case when this matters. In 99% of cases master request was ok and there were zero chance of anything going wrong.

Myself I'm not too familiar with FOS JS routing either, so no strong opinion there, just wanted to point out a potential issue that goes under the radar.

...and back to the bug itself.

Since the command apparently was working fine in newer kernel versions (where we had `RequestStack` injected) without the check against `getMasterRequest()` returning `null`, I assume that `getBaseUrl()` is never called in CLI context. I'd be glad for a check (with exception or fallback), but It's just a nice-to-have from my POV.